pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -487,75 +460,109 @@ def _build_circbox(instr: Instruction, circuit: QuantumCircuit) -> CircBox:
 # docs -> https://docs.quantum.ibm.com/api/qiskit/qiskit.circuit.IfElseOp
 # Examples -> https://docs.quantum.ibm.com/guides/classical-feedforward-and-control-flow
 # pytket-qiskit issue -> https://github.com/CQCL/pytket-qiskit/issues/415
-def _pytket_boxes_from_ifelseop(
-    if_else_op: IfElseOp, qregs: list[QuantumRegister], cregs: list[ClassicalRegister]
-) -> tuple[CircBox, Optional[CircBox]]:
+def _pytket_circuits_from_ifelseop(
+    if_else_op: IfElseOp,
+    outer_builder: "CircuitBuilder",
+    qargs: list[QCQubit],
+    cargs: list[Clbit],
+) -> tuple[Circuit, Optional[Circuit]]:
     # Extract the QuantumCircuit implementing true_body
     if_qc: QuantumCircuit = if_else_op.blocks[0]
-    if_builder = CircuitBuilder(qregs, cregs)
+    # if_qc can have empty qregs, so build from bits
+    if_builder = CircuitBuilder.from_bits(
+        if_qc.qubits, if_qc.clbits, if_qc.name, param_to_tk(if_qc.global_phase)
+    )
     if_builder.add_qiskit_data(if_qc)
     if_circuit = if_builder.circuit()
+    # if_circuit might have a different set of registers, which might
+    # cause problems when appending to the outer circuit.
+    # We rename the units to make sure the registers in the inner circuit
+    # is a subset of the registers in the ourter circuit.
+    rename_map = {}
+    for i, inner_q in enumerate(if_qc.qubits):
```

Comment:
This logic is repeated for the "else block". Could make ahelper. What do you think?